feat: adds get/set config functions

mdarrik opened the pull request:
<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

## WHY are these changes introduced?

Addresses [ENG-7167](https://nacelle.atlassian.net/browse/ENG-7167) - Adds a setConfig and getConfig to the cart client. <!-- link to Jira or Github issue if one exists -->

<!--
  Context about the problem that’s being addressed. Use bullets or ordered lists for multiple touch points.
-->

## WHAT is this pull request doing?

- Adds a getConfig method that returns the string options passed to `createCartClient`
- Adds a `setConfig` method to allow you to update the `language` and `countryCode`
<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

## How to Test

- Checkout this branch `mdarrik/shopify-cart/ENG-7167-add-setConfig-properties-to-cart` 
- Run `npm run test:ci` - all tests should pass
- Run `npm run build` - should build successfully
- To test in a local repo - use `npm link` and try calling `getConfig` to see the options used in the cart, and test updating the language/country with `setConfig`

NWRichmond reviewed (commented):
The `build` and `test:ci` scripts are working as expected ✅

Running the `codegen` script results in uncommitted changes, but they all appear to be changes on Shopify's end (adding periods to the end of comments), and not related to changes made in this PR. Feel free to run `codegen` again and commit changes, or we can handle that on the [`ENG-6843/shopify-cart-1.0.0`](https://github.com/getnacelle/nacelle-js/tree/ENG-6843/shopify-cart-1.0.0) branch.

The `setConfig` is setting the `country` and `language` config as expected:

https://user-images.githubusercontent.com/5732000/191074102-ea3d0521-3985-456f-b161-4e19ed293389.mov

Personally, I find the results of `getConfig` somewhat unintuitive. I think it might make more sense to only return properties from `getConfig` that are settable in `setConfig`. I think it'd be perfectly fine to remove `shopifyShopId` from `shopifyStorefrontAccessToken` from the `getConfig` results, as this is something that we probably shouldn't expect devs to need to check in their code.

mdarrik commented:
> Personally, I find the results of `getConfig` somewhat unintuitive. I think it might make more sense to only return properties from `getConfig` that are settable in `setConfig`. I think it'd be perfectly fine to remove `shopifyShopId` from `shopifyStorefrontAccessToken` from the `getConfig` results, as this is something that we probably shouldn't expect devs to need to check in their code.

Yeah. I was sort of conflicted on this part of the `getConfig` piece. My thought on keeping it was that it provides a way to provide debugging/setup information which could be useful for merchant developers trying to debug setup, or for merchants with multiple shopify storefronts to cover their different locals. But happy to drop those fields if we think that's not all that useful/beneficial.

NWRichmond commented:
Ah, good to know. I think removing them from the `getConfig` response is probably the right move for now. We can always reevaluate that in response to feature requests etc.

> or for merchants with multiple shopify storefronts to cover their different locales

For what it's worth, because of the nature of [Shopify Markets](https://help.shopify.com/en/manual/markets), merchants should be able to accomplish international support via a single Shopify store (& token).

mdarrik commented:
@NWRichmond removed those fields from `getConfig`. Also updated the docs to include both methods. 

While I was in there I added an href to all of the method `a` tags so they are [valid links](https://twitter.com/BenDMyers/status/1571849889155260417). If we don't want them to be, we should switch them to spans instead.

NWRichmond reviewed (approved):
Awesome! Thanks for those updates, @mdarrik. Everything is working as expected.

![shopify cart getConfig returns language and country](https://user-images.githubusercontent.com/5732000/191376363-a08a9d58-aa1f-4793-90c7-2b086e3c03e8.png)
